src/main/java/seedu/address/model/graduation/FocusAreaGraduationRequirement.java

```diff
@@ -5,40 +5,55 @@
 import seedu.address.model.Model;
 import seedu.address.model.module.ModuleCode;
 import seedu.address.model.programmes.specialisations.GenericSpecialisation;
+import seedu.address.model.programmes.specialisations.cs.GenericCsSpecialisation;
 
 public class FocusAreaGraduationRequirement extends GraduationRequirement {
 
-    private GenericSpecialisation genericSpecialisation = null;
+    private GenericSpecialisation specialisation = null;
 
     public FocusAreaGraduationRequirement(Model model) {
         try {
-            genericSpecialisation = model.getActiveStudent().getSpecialisation();
+            specialisation = model.getActiveStudent().getSpecialisation();
         } catch (Exception e) {
             return;
         }
     }
 
-    public void setGenericSpecialisation(GenericSpecialisation genericSpecialisation) {
-        this.genericSpecialisation = genericSpecialisation;
+    public void setSpecialisation(GenericSpecialisation specialisation) {
+        this.specialisation = specialisation;
     }
 
     public boolean isFulfilled(List<ModuleCode> moduleCodes) {
-        if (genericSpecialisation == null) {
+        if (specialisation == null) {
             return false;
         }
-        return genericSpecialisation.isFulfilled(moduleCodes);
+        return specialisation.isFulfilled(moduleCodes);
     }
 
-    public GenericSpecialisation getGenericSpecialisation() {
-        return genericSpecialisation;
+    public GenericSpecialisation getSpecialisation() {
+        return specialisation;
     }
 
     public String getString(List<ModuleCode> moduleCodes) {
         try {
-            return "[" + getStatusIcon(isFulfilled(moduleCodes)) + "] ";
-            // return "[" + getStatusIcon(isFulfilled(moduleCodes)) + "] " + genericSpecialisation.toString();
+            StringBuilder sb = new StringBuilder()
+                .append("[").append(getStatusIcon(specialisation.isFulfilled(moduleCodes))).append("] " + "Focus Area: ").append(specialisation.getName())
+                .append("\n    [").append(getStatusIcon(((GenericCsSpecialisation) specialisation).arePrimariesFulfilled(moduleCodes))).append("] Primaries");
+            for (ModuleCode primaries : ((GenericCsSpecialisation) getSpecialisation()).getPrimaries()) {
+                sb.append("\n        ").append(new SingleGraduationRequirement(primaries).getString(moduleCodes));
+            }
+            sb.append("\n    [").append(getStatusIcon(((GenericCsSpecialisation) specialisation).areElectivesFulfilled(moduleCodes))).append("] Electives");
+            for (ModuleCode electives : ((GenericCsSpecialisation) getSpecialisation()).getElectives()) {
```

Comment:
Not sure if checking that `specialisation` is an instance of `GenericCsSpecialisation` is necessary.

Reply:
@thetruevincentchow I believe he is casting,  and not checking. Because ```specialisation``` does not have the methods he's trying to use